keras_nlp/models/bert/bert_models.py

```diff
@@ -226,13 +227,43 @@ def presets(cls):
         return copy.deepcopy(backbone_presets)
 
     @classmethod
+    @format_docstring(names=", ".join(backbone_presets))
     def from_preset(
         cls,
         preset,
         load_weights=True,
         **kwargs,
     ):
-
+        """Instantiate BERT model from preset architecture and weights.
+
+        Args:
+            preset: string. Must be one of {names}.
+            load_weights: Whether to load pre-trained weights into model.
+                Defaults to `True`.
+
+        Examples:
+        ```python
+        input_data = {{
```

Comment:
Why the double {?

Reply:
Because this delegates to str.format, and str.format require escaping `{` as `{{`. I should definitely document that. But if you can think of a clever way around it, let me know!

Reply:
Oh interesting. Given that dicts are super common in code examples in docstrings, this is a major inconvenience. I would argue for dropping the `format()` API convention and instead adopting the Jinja2 API convention, i.e. the reverse:

- `{thing}` gets rendered as is
- `{{thing}}` gets rendered as the value for the variable "thing"

Reply:
Yeah, this is a good idea. We can do this by simple swapping {{ and { before calling format.